### PR TITLE
Badge component

### PR DESCRIPTION
**Pull request by shaswat-indian**
This PR adds the badge component.

https://github.com/reaviz/reablocks/assets/33908100/f9b23f7b-43a3-4902-88d0-0ce3fb62c44e

### CODE REVIEW

**Comment by amcdnl** [comment]
Something odd is going on - the bubble should have the color.

**Comment by shaswat-indian** [comment]
> Something odd is going on - the bubble should have the color.

Updated with colors for the bubbles.

https://github.com/reaviz/reablocks/assets/33908100/854fbac3-c233-476d-b40c-7678c0a896f2

File: src/elements/Badge/Badge.module.css [hunk 1]
```diff
@@ -21,7 +21,7 @@
   box-sizing: border-box;
   line-height: 1;
   padding: var(--spacing-xs);
-  border-radius: var(--border-radius-md);
+  border-radius: 50%;
```
Comment (amcdnl) [inline src/elements/Badge/Badge.module.css:24]: This should be variable based.

File: src/utils/Theme/themes/dark.ts [hunk 2]
```diff
@@ -411,6 +411,16 @@ export const darkTheme: Theme = {
     },
     calendar: {
       'calendar-spacing': '8px'
+    },
+    badge: {
+      'badge-color-default': darkColors.white,
+      'badge-color-on-default': darkColors.black,
```
Comment (amcdnl) [inline src/utils/Theme/themes/dark.ts:417]: We don't use this `on` pattern in the other spots, let's match that style please.

**Review by amcdnl — COMMENTED** [review]
Also need to add a docs page like the other components and export the component in the index.

File: src/elements/Badge/Badge.module.css (hunk 1 above)
Comment (shaswat-indian) [reply src/elements/Badge/Badge.module.css]: Added a new token.

File: src/utils/Theme/themes/dark.ts (hunk 2 above)
Comment (shaswat-indian) [reply src/utils/Theme/themes/dark.ts]: Updated but I saw that pattern at [another place](https://github.com/reaviz/reablocks/blob/4f8943f22512564be66ce8836d7e3ba11e04d7f0/src/utils/Theme/themes/dark.ts#L219-L220) as well.

**Comment by shaswat-indian** [comment]
> Also need to add a docs page like the other components and export the component in the index.

Exported and added docs page.

**Review by amcdnl — APPROVED** [review]

**Pull request merged**